cmd/controller/main.go

```diff
@@ -2,42 +2,41 @@ package main
 
 import (
 	"crypto/rand"
-	"crypto/rsa"
 	"crypto/x509"
-	"crypto/x509/pkix"
 	goflag "flag"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"log"
-	"math/big"
 	"os"
 	"os/signal"
 	"strings"
 	"syscall"
 	"time"
 
 	flag "github.com/spf13/pflag"
-	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	certUtil "k8s.io/client-go/util/cert"
 
 	sealedsecrets "github.com/bitnami-labs/sealed-secrets/pkg/client/clientset/versioned"
 	ssinformers "github.com/bitnami-labs/sealed-secrets/pkg/client/informers/externalversions"
 )
 
 var (
-	keyName      = flag.String("key-name", "sealed-secrets-key", "Name of Secret containing public/private key.")
-	keySize      = flag.Int("key-size", 4096, "Size of encryption key.")
-	validFor     = flag.Duration("key-ttl", 10*365*24*time.Hour, "Duration that certificate is valid for.")
-	myCN         = flag.String("my-cn", "", "CN to use in generated certificate.")
-	printVersion = flag.Bool("version", false, "Print version information and exit")
+	keyLabelName    = flag.String("key-label", "sealed-secrets-key", "Label used to identify public/private key pairs in k8s.")
+	keyPrefix       = flag.String("key-prefix", "", "Prefix used to name keys. Defaults to label name.")
+	keySize         = flag.Int("key-size", 4096, "Size of encryption key.")
+	validFor        = flag.Duration("key-ttl", 10*365*24*time.Hour, "Duration that certificate is valid for.")
+	myCN            = flag.String("my-cn", "", "CN to use in generated certificate.")
+	printVersion    = flag.Bool("version", false, "Print version information and exit")
+	keyRotatePeriod = flag.Duration("rotate-period", 30*24*time.Hour, "New key generation period")
 
 	// VERSION set from Makefile
 	VERSION = "UNKNOWN"
+
+	// Selector used to find existing public/private key pairs on startup
+	keySelector = SealedSecretsKeyLabel + "=" + "active"
```

Comment:
If you want to be fancy, this could be:
```go
import "k8s.io/apimachinery/pkg/fields"

keySelector = fields.OneTermEqualSelector(SealedSecretsKeyLabel, "active")
```

And then call `.String()` on it when building the actual k8s label query.  This would be particularly useful if including another selector (perhaps from a command line flag), since there are functions in that package for building And queries, etc.

Reply:
Thanks for the tip 👍